entity_gym/entity_gym/envs/griddly/enn_wrapper.py

```diff
@@ -0,0 +1,193 @@
+        self._env.reset()
+        self._current_g_state = self._env.get_state()
+        self._entity_names = self._env.object_names
+
+        self._obs_space = self._generate_obs_space()
+        self._action_space = self._generate_action_space()
+
+    def _generate_obs_space(self) -> ObsSpace:
+
+        # TODO: currently we flatten out all possible variables regardless of entity.
+        # Each entity contains x, y, z positions, plus the values of all variables
+        # TODO: need a Griddly API which tells us which variables are for each entity
+        # TODO: need a Griddly API to get the names of global variables
+        global_variables = list(self._current_g_state["GlobalVariables"].keys())
+
+        # Global entity for global variables and global actions (these dont really exist in Griddly)
+        space = {"__global__": Entity(global_variables)}
+        for name in self._entity_names:
+            space[name] = Entity(
+                ["x", "y", "z", "orientation", "player_id", *self._env.variable_names]
+            )
+
+        return ObsSpace(space)
+
+    def _to_griddly_action(self, action: Mapping[str, Action]) -> np.ndarray:
+
+        for action_name, a in action.items():
+            action_type = self._env.action_names.index(action_name)
+            # TODO: this only works if we have a single entity, otherwise we have to map the entityID to an x,y coordinate
+            action_id = a.actions[0][1]
+
+        return np.array([action_type, action_id])
+
+    def _generate_action_space(self) -> Dict[str, ActionSpace]:
+
+        action_space: Dict[str, ActionSpace] = {}
+        for action_name, action_mapping in self._env.action_input_mappings.items():
+            # Ignore internal actions for the action space
+            if action_mapping["Internal"] == True:
+                continue
+
+            input_mappings = action_mapping["InputMappings"]
+
+            actions = []
+            actions.append("NOP")  # In Griddly, Action ID 0 is always NOP
+            for action_id in range(1, len(input_mappings) + 1):
+                mapping = input_mappings[str(action_id)]
+                description = mapping["Description"]
+                actions.append(description)
+
+            action_space[action_name] = CategoricalActionSpace(actions)
+
+        return action_space
+
+    def _get_entity_observation(self) -> Dict[str, np.ndarray]:
+        self._current_g_state = self._env.get_state()
+
+        def orientation_feature(orientation_string: str) -> int:
+            if orientation_string == "NONE":
+                return 0
+            elif orientation_string == "UP":
+                return 0
+            elif orientation_string == "RIGHT":
+                return 1
+            elif orientation_string == "DOWN":
+                return 2
+            elif orientation_string == "LEFT":
+                return 3
+            else:
+                raise RuntimeError("Unknown Orientation")
```

Comment:
I'd replace this with a dict lookup, somewhat more concise and probably faster as well.

Reply:
This will be pushed into c++ , so these inefficiencies will disappear fairly quickly